### PR TITLE
Add SupplyPartnerId to the AdyenPayments request

### PR DESCRIPTION
**Pull request by mohadian**
## JIRA
[MOB-3205](https://karhoo.atlassian.net/browse/MOB-3205)

## DESCRIPTION
Add SupplyPartnerId to the AdyenPayments request

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [X] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [X] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [X] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift [hunk 1]
```diff
@@ -36,11 +39,38 @@ final class AdyenCardRegistrationFlowSpec: XCTestCase {
         testObject.setBaseView(mockBaseViewController)
     }
 
+    func testStartingDropIn() {
```
Comment (jsm20) [inline KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift:42]: Do we need this?

File: KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift [hunk 2]
```diff
@@ -36,11 +39,38 @@ final class AdyenCardRegistrationFlowSpec: XCTestCase {
...
+        switch result {
+        case .completed(let result): cardRegistrationFlowCompletionResult = result
+        case .cancelledByUser: cardRegistrationCancelledByUser = true
+        }
     }
 
-    func testStartingDropIn() {
+    /**
+      * Given: The flow starts
+      * When: Currency and Amont is provided
```
Comment (jsm20) [inline KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift:55]: Picky comment but spelling of amount 😆

File: KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift [hunk 3]
```diff
@@ -36,11 +39,38 @@ final class AdyenCardRegistrationFlowSpec: XCTestCase {
...
+      * Then: AdyenPaymentMethodsRequest is made with the correct payload
+      */
+    func testFlowStart() {
+        simulateStartFlow()
+
+        XCTAssert(mockPaymentService.adyenPaymentMethodsCalled)
+        XCTAssertEqual(amount, mockPaymentService.adyenPaymentMethodsRequest?.amount?.value)
+        XCTAssertEqual(mockCardCurrencyCode, mockPaymentService.adyenPaymentMethodsRequest?.amount?.currency)
```
Comment (jsm20) [inline KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift:63]: Should we assert the supplier id on the request?

Comment (mohadian) [reply KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift:63]: The supplier ID is populated as part of the AdyenPaymentsRequest.

File: KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift (hunk 1 above)
Comment (mohadian) [reply KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift:42]: Removed

File: KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift (hunk 3 above)
Comment (mohadian) [reply KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift:60]: the amount and currency are part of `adyenPaymentMethods` and supplierPartnerId is part of `adyenPayments` ... the first one is made as soon as the flow starts ... the other one is made later during the process

**Review by jsm20 — APPROVED** [review]

**Pull request merged**